test/v4/controllers/RetrieveCalculationControllerSpec.scala

```diff
@@ -165,19 +189,37 @@ class RetrieveCalculationControllerSpec
     }
   }
 
-  private trait Test extends ControllerTest {
+  private trait Test extends ControllerTest with AuditEventChecking {
 
-    private lazy val controller = new RetrieveCalculationController(
+    lazy val controller = new RetrieveCalculationController(
       authService = mockEnrolmentsAuthService,
       lookupService = mockMtdIdLookupService,
       validatorFactory = mockRetrieveCalculationValidatorFactory,
       service = mockRetrieveCalculationService,
       cc = cc,
       hateoasFactory = mockHateoasFactory,
-      idGenerator = mockIdGenerator
+      idGenerator = mockIdGenerator,
+      auditService = mockAuditService
     )
 
-    protected def callController(): Future[Result] = controller.retrieveCalculation(nino, taxYear, calculationId)(fakeRequest)
+    protected def callController(): Future[Result] =
+      controller.retrieveCalculation(nino, taxYear, calculationId)(fakeRequest)
+
+    protected def event(auditResponse: AuditResponse, requestBody: Option[JsValue] = None): AuditEvent[GenericAuditDetail] =
+      AuditEvent(
+        auditType = "RetrieveATaxCalculation",
+        transactionName = "retrieve-a-tax-calculation",
+        detail = GenericAuditDetail(
+          versionNumber = "4.0",
+          userType = "Individual",
+          agentReferenceNumber = None,
+          params = Map("nino" -> nino, "calculationId" -> calculationId, "taxYear" -> taxYear),
+          requestBody = None,
```

Comment:
Otherwise individual tests cannot assert the audited request body value properly:
```suggestion
          requestBody = requestBody,
```